Content.Server/NPC/Systems/Faction/FactionSystem.cs

```diff
@@ -44,6 +51,47 @@ private void OnFactionStartup(EntityUid uid, FactionComponent component, Compone
             RefreshFactions(component);
         }
 
+        private void OnGetNearbyHostiles(EntityUid uid, FactionComponent component, ref GetNearbyHostilesEvent args)
+        {
+            args.ExceptionalFriendlies.UnionWith(component.ExceptionalFriendlies);
+        }
+
+        /// <summary>
+        /// If we bought something we probably don't want it to start biting us after it's automatically placed in our hands.
+        /// If you do, consider finding a better solution to grenade penguin CBT.
+        /// </summary>
+        private void OnPurchased(EntityUid uid, FactionComponent component, ref ItemPurchasedEvent args)
+        {
+            component.ExceptionalFriendlies.Add(args.Purchaser);
+        }
+
+        private void OnEquipped(EntityUid uid, ClothingAddFactionComponent component, GotEquippedEvent args)
+        {
+            if (!TryComp<ClothingComponent>(uid, out var clothing))
+                return;
+
+            if (!clothing.Slots.HasFlag(args.SlotFlags))
+                return;
+
+            if (!TryComp<FactionComponent>(args.Equipee, out var factionComponent))
+                return;
+
+            if (factionComponent.Factions.Contains(component.Faction))
+                return;
```

Comment:
I think AddFaction already handles duplicates since `factionComponent`.Factions is a hashset

Reply:
this isn't testing for duplicates it's to prevent us from activating the component